shared/js/background/declarative-net-request.js

```diff
@@ -35,14 +38,17 @@ const ruleIdRangeByConfigName = {
 // User allowlisting and the ServicerWorker initiated request exception both
 // only require one declarativeNetRequest rule, so hardcode the rule IDs here.
 export const USER_ALLOWLIST_RULE_ID = 20001
-export const SERVICE_WORKER_INITIATED_ALLOWING_RULE_ID = 20002
 export const ATB_PARAM_RULE_ID = 20003
 const RESERVED_RULE_IDS = [
     USER_ALLOWLIST_RULE_ID,
-    SERVICE_WORKER_INITIATED_ALLOWING_RULE_ID,
     ATB_PARAM_RULE_ID
 ]
 
+// Rule IDs for static session rules
+export const SERVICE_WORKER_INITIATED_ALLOWING_RULE_ID = 100
```

Comment:
My vote would be to keep these in the reserved rule ID range, and to get rid of `clearInvalidRules` + `RESERVED_RULE_IDS`. I haven't checked if session/dynamic rule IDs can overlap, but it seems kind of confusing to have them overlap even if it's allowed IMO.

Reply:
We previously didn't have the `ruleIdRangeByConfigName['_RESERVED']` range (20001 to 21000), but we had some hard-coded rule IDs in that range (e.g. `USER_ALLOWLIST_RULE_ID` with ID of 20001). That was a "foot-gun waiting to happen", since when a new config rule ID range got added (combined) the logical place to put it was (20001, 30000) and that then immediately clashed with the hard-coded rule IDs.

Sam and I went back-and-forth on this a few times, weighing changing the IDs around vs making things clear. We settled on adding the "_RESERVED" range to make it explicit without changing the rule IDs for any existing installs.

Along the way, Sam added the `RESERVED_RULE_IDS` and `clearInvalidRules` to try to and clear rules with the wrong IDs. The idea there was to fix installs that were broken by us changing the rule IDs about, but since that didn't happen in the end IMO that logic should be removed. This PR makes me feel stronger to remove it, since it's getting kind of confusing with session VS dynamic rules and this doesn't help.